tools/bin/hawqregister

```diff
@@ -40,186 +39,195 @@ EXECNAME = os.path.split(__file__)[-1]
         sys.exit(1)
 
 
-def get_seg_name(options, databasename, tablename):
+def get_seg_name(dburl, tablename, database, fmt):
     try:
-        relfilenode = 0
-        relname = ""
-        query = ("select pg_class2.relname from pg_class as pg_class1, pg_appendonly, pg_class as pg_class2 where pg_class1.relname ='%s' "
-                 "and  pg_class1.oid = pg_appendonly.relid and pg_appendonly.segrelid = pg_class2.oid;") % tablename
-        dburl = dbconn.DbURL(hostname=options.host, port=options.port, username=options.user, dbname=databasename)
+        relname = ''
+        tablename = tablename.split('.')[-1]
+        query = ("select pg_class2.relname from pg_class as pg_class1, pg_appendonly, pg_class as pg_class2 "
+                 "where pg_class1.relname ='%s' and pg_class1.oid = pg_appendonly.relid and pg_appendonly.segrelid = pg_class2.oid;") % tablename
         conn = dbconn.connect(dburl, True)
         rows = dbconn.execSQL(conn, query)
-	conn.commit()
-        if rows.rowcount == 0:
-            logger.error("table '%s' not found in db '%s'" % (tablename, databasename));
+        conn.commit()
+        if not rows.rowcount:
+            logger.error('table "%s" not found in db "%s"' % (tablename, database))
             sys.exit(1)
         for row in rows:
             relname = row[0]
         conn.close()
-
     except DatabaseError, ex:
-        logger.error("Failed to connect to database, this script can only be run when the database is up")
-        logger.error("host = %s, port = %d, user = %s, dbname = %s, query = %s" % (options.host, options.port, options.user, databasename, query))
-        sys.exit(1)
-
-    # check whether the target table is parquet format
-    if relname.find("paq") == -1:
-        logger.error("table '%s' is not parquet format" % tablename)
+        logger.error('Failed to run query "%s" with dbname "%s"' % (query, database))
         sys.exit(1)
+    if fmt == 'Parquet':
+        if relname.find("paq") == -1:
+            logger.error("table '%s' is not parquet format" % tablename)
+            sys.exit(1)
 
     return relname
 
 
-def check_hash_type(options, databasename, tablename):
+def check_hash_type(dburl, tablename):
+    '''Check whether target table is hash-typed, in that case simple insertion does not work'''
     try:
         query = "select attrnums from gp_distribution_policy, pg_class where pg_class.relname = '%s' and pg_class.oid = gp_distribution_policy.localoid;" % tablename
-        dburl = dbconn.DbURL(hostname=options.host, port=options.port, username=options.user, dbname=databasename)
         conn = dbconn.connect(dburl, False)
         rows = dbconn.execSQL(conn, query)
-	conn.commit()
-        if rows.rowcount == 0:
-            logger.error("target not found in table gp_distribution_policy")
+        conn.commit()
+        if not rows.rowcount:
+            logger.error('Target not found in table gp_distribution_policy.')
```

Comment:
Will this thrown out to user? If yes, we need to modify the warning content. What's the meaning of 'Target'?